From 88cce5c6ce3689a989faf7f383332653706c65c3 Mon Sep 17 00:00:00 2001
From: Jun Nie <jun.nie@linaro.org>
Date: Mon, 28 May 2018 11:36:41 +0800
Subject: [PATCH 09/24] warp7: Build dtb into u-boot

Add u-boot.dtb so that key for signature can be
stored in it.

Signed-off-by: Jun Nie <jun.nie@linaro.org>
---
 arch/arm/dts/Makefile        |  3 ++-
 arch/arm/dts/imx7s-warp7.dts | 11 +++++++++++
 configs/warp7_bl33_defconfig |  2 ++
 3 files changed, 15 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm/dts/imx7s-warp7.dts

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 44ebc50bfa..1e893e1b74 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -448,7 +448,8 @@ dtb-$(CONFIG_MX6ULL) += imx6ull-14x14-evk.dtb
 
 dtb-$(CONFIG_MX7) += imx7-colibri.dtb \
 	imx7d-sdb.dtb \
-	imx7d-sdb-qspi.dtb
+	imx7d-sdb-qspi.dtb \
+	imx7s-warp7.dtb
 
 dtb-$(CONFIG_ARCH_MX7ULP) += imx7ulp-evk.dtb
 
diff --git a/arch/arm/dts/imx7s-warp7.dts b/arch/arm/dts/imx7s-warp7.dts
new file mode 100644
index 0000000000..ad2da8ca6e
--- /dev/null
+++ b/arch/arm/dts/imx7s-warp7.dts
@@ -0,0 +1,11 @@
+/*
+ * Copyright 2018 Linaro.ltd
+ *
+ */
+
+/dts-v1/;
+
+/* Add an empty dts so that fit signature can be stored in dtb later */
+/ {
+	model = "Warp7 iMX7S";
+};
diff --git a/configs/warp7_bl33_defconfig b/configs/warp7_bl33_defconfig
index 41acb004d5..d1befc42dc 100644
--- a/configs/warp7_bl33_defconfig
+++ b/configs/warp7_bl33_defconfig
@@ -43,3 +43,5 @@ CONFIG_OPTEE_TZDRAM_SIZE=0x2000000
 CONFIG_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_FIT_VERBOSE=y
+CONFIG_DEFAULT_DEVICE_TREE="imx7s-warp7"
+CONFIG_OF_CONTROL=y
-- 
2.23.0.rc1

